scipy/interpolate/tests/test_interpolate.py

```diff
@@ -2744,6 +2744,25 @@ def test_length_one_axis2(self, fill_value, method):
         else:
             assert_allclose(zb, fill_value)
 
+    def test_nan_x_1d(self):
+        # gh-6624 : if x is nan, result should be nan
+        f = RegularGridInterpolator(([1, 2, 3],), [10, 20, 30],
```

Comment:
```suggestion
        f = RegularGridInterpolator(([1, 2, 3],), [10, 20, 30], fill_value=1,
```
to show that this isn't just passing because the default fill values is `np.nan`.

Reply:
Thanks! There is a test with nans and non-nans below, but it wouldn't hurt to test it more.

Reply:
Yeah I saw. It's just a reflex from working with stats, as there have been a lot of issues with nans in variable numbers of dimensions.